engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -1494,37 +1494,49 @@ private int validateConstraintsForGroup(ValidationContext<?> validationContext,
 		BeanMetaData<?> metaData = beanMetaDataManager.getBeanMetaData( clazz );
 		PropertyMetaData property = metaData.getMetaDataFor( elem.getName() );
 
-		// use precomputed method list as ReflectionHelper#containsMember is slow
 		if ( property == null ) {
 			throw log.getInvalidPropertyPathException( elem.getName(), metaData.getBeanClass() );
 		}
-		else if ( !propertyIter.hasNext() ) {
+
+		// if we are in the case of an iterable and we want to validate an element of this iterable, we have to get the
+		// element value
+		if ( property.isCascading() && elem.isIterable() && propertyIter.hasNext() ) {
+			elem = (NodeImpl) propertyIter.next();
+			newValue = newValue == null ? null : getValue( newValue, validationContext, property );
+
+			if ( newValue != null && elem.getIndex() != null ) {
+				newValue = ReflectionHelper.getIndexedValue( newValue, elem.getIndex() );
+			}
+			else if ( newValue != null && elem.getKey() != null ) {
+				newValue = ReflectionHelper.getMappedValue( newValue, elem.getKey() );
+			}
+			else if ( newValue != null ) {
+				throw log.getPropertyPathMustProvideIndexOrMapKeyException();
+			}
+			clazz = (Class<?>) ReflectionHelper.getIndexedType( property.getType() );
```

Comment:
Instead of `property.getType()` the actual value type should be used (in the case of `validateProperty()`, as BV dicates to use runtime type rather than static type for constraint retrieval.